src/lib.rs

```diff
@@ -475,19 +558,60 @@ impl Sender {
     /// is greater than the ring buffer capacity, this method will return
     /// `Err(`[`Error::MessageTooLarge`](enum.Error.html#variant.MessageTooLarge)`))`.
     pub fn send_when_empty(&self, value: &impl Serialize) -> Result<()> {
+        unsafe { self.send_timeout_0(&BincodeShmWriter(value), true, None).map(drop) }
+    }
+
+    /// Send the specified message, waiting for sufficient contiguous space to become available in the ring buffer
+    /// if necessary. Uses the lower level `ShmWriter` trait to write messages directly into the mmap buffer. Is 
+    /// inherently unsafe due to complete control it gives over writing to memory.
+    ///
+    /// The size of the message must be greater than zero or else this method will return
+    /// `Err(`[`Error::ZeroSizedMessage`](enum.Error.html#variant.ZeroSizedMessage)`))`.  If the message size is
+    /// greater than the ring buffer capacity, this method will return
+    /// `Err(`[`Error::MessageTooLarge`](enum.Error.html#variant.MessageTooLarge)`))`.
+    pub unsafe fn send_direct(&self, value: &impl ShmWriter) -> Result<()> {
```

Comment:
I know I recommended adding `unsafe` to these, but now that `ShmWriter::write_to_shim` takes a `&mut [u8]` instead of a `*mut u8`, I think we can remove the `unsafe` keywords from the `send_direct*` methods.  AFAICT, there's no way for the caller to use them unsoundly unless the caller (or `ShmWriter` implementer) itself uses `unsafe`.  Sorry for the confusion.